tests/jballerina-unit-test/src/test/resources/test-src/object/object-equivalency-01-negative.bal

```diff
@@ -401,4 +401,76 @@ function testObjectMemberOrder() returns [PersonInOrder, PersonNotInOrder] {
     PersonInOrder p4 = p3;
 
     return [p4, p2];
-}
\ No newline at end of file
+}
+
+type ObjWithOnlyRemoteMethod client object {
+    public string name;
+    public string id = "";
+
+    function __init(string name) {
+        self.name = name;
+    }
+    public remote function send(string message) returns error? {
+    }
+    public remote function receive(string message) {
+    }
+};
+
+type ObjWithRemoteMethod client object {
+    public string name;
+    public string id = "";
+
+    function __init(string name) {
+        self.name = name;
+    }
+    public remote function send(string message) returns error? {
+    }
+    public function receive(string message) {
+    }
+};
+
+type NonClientObj object {
+    public string name;
+    public string id = "";
+
+    function __init(string name) {
+        self.name = name;
+    }
+    public function send(string message) returns error? {
+    }
+    public function receive(string message) {
+    }
+};
+
+type ClientObjWithoutRemoteMethod client object {
+    public string name;
+    public string id = "";
+
+    function __init(string name) {
+        self.name = name;
+    }
+    public function send(string message) returns error? {
+    }
+    public function receive(string message) {
+    }
+};
+
+function testEqOfObjectsWithAndWithoutRemoteMethods1() {
+    NonClientObj e = new ("email-1");
+    ObjWithRemoteMethod p = e;
+}
+
+function testEqOfObjectsWithAndWithoutRemoteMethods2() {
+    ClientObjWithoutRemoteMethod e = new ("email-2");
+    ObjWithRemoteMethod p = e;
+}
+
+function testEqOfObjectsWithAndWithoutRemoteMethods3() {
+    ClientObjWithoutRemoteMethod e = new ("email-3");
+    ObjWithOnlyRemoteMethod = e;
+}
+
+function testEqOfObjectsWithAndWithoutRemoteMethods4() {
+    NonClientObj e = new ("email-4");
+    ObjWithOnlyRemoteMethod = e;
```

Comment:
Aren't all these tests testing the source without remote and the target with? Can we add the opposite too?

Reply:
Done